solutions/src/main/java/ru/sberbank/school/task02/Calculator.java

```diff
@@ -0,0 +1,50 @@
+package ru.sberbank.school.task02;
+
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+public class Calculator implements FxConversionService {
+    private ExternalQuotesService provider;
+
+    Calculator(ExternalQuotesService externalQuotesService) {
+        this.provider = externalQuotesService;
+    }
+
+    @Override
+    public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal amount) {
+        List<Quote> quotes = provider.getQuotes(symbol);
+        quotes.sort((right, left)-> {
+            if (left.isInfinity()) return 1;
+            if (right.isInfinity()) return -1;
+            return left.getVolumeSize().compareTo(right.getVolumeSize());
+        });
+        BigDecimal currentOffer = BigDecimal.ZERO;
+        for (Quote quote: quotes) {
+            if (quote.isInfinity()) {
+                currentOffer = quote.getOffer();
+            } else if (amount.compareTo(quote.getVolumeSize()) < 0) {
+                currentOffer = quote.getOffer();
+            } else {
+                break;
+            }
+        }
+        return currentOffer;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
&#39;if&#39; construct must use &#39;{}&#39;s.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a654561505>*